src/inmanta/protocol/rest/__init__.py

```diff
@@ -665,11 +668,10 @@ async def _execute_call(
             # Authorization might need data from the request but we do not want to process it before we are sure the call
             # is authenticated.
             arguments = CallArguments(config, message, request_headers)
-            arguments.authenticate(server_config.server_enable_auth.get())
+            is_auth_enabled: bool = self.is_auth_enabled()
+            arguments.authenticate(auth_enabled=is_auth_enabled)
```

Comment:
Am I correct that `is_auth_enabled()` is equivalent to `self.get_policy_engine() is not None`? In that case, we could drop the method altogether at this level (and get rid of the `NotImplementedError`, and simply use `policy_engine is not None` here?

Reply:
It's not really the same. In the future we will have two modes for authorization. One legacy mode, which doesn't use the policy engine but just validates the token. An one where the policy engine is used. This implementation is not yet there because the PR was becoming too big in my opinion. I'll add that in a follow up PR. To summarize, it's possible to have authentication enabled, while the authorization doesn't use the policy engine.